### PR TITLE
Support custom labels on the driver pod.

### LINKED ISSUE #26: Support custom labels for the driver pod
One example use case is the driver pod also serves content, and the user who submits the driver wants to expose that content as a service. Instead of needing to worry about what labels spark-submit will choose, they can just provide their own labels (on top of the ones Spark generates) and have the service look for pods with said custom labels.

### PR DESCRIPTION
**Pull request by mccheah**
Closes #26.

### CODE REVIEW

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -213,6 +213,14 @@ from the other deployment modes. See the [configuration page](configuration.html
     (typically 6-10%).
   </td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.driver.labels</code></td>
+  <td>(none)</td>
+  <td>
+    Custom labels that will be added to the driver pod. This should be a comma-separated list of label key-value pairs,
+    where each label is in the format <code>&ltkey&gt=&ltvalue&gt</code>.
```
Comment (ash211) [inline docs/running-on-kubernetes.md]: get rid of the &lt -- I think just `<code>key=value</code>` is clear enough

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala [hunk 2]
```diff
@@ -184,7 +188,7 @@ private[spark] class Client(
           kubernetesClient.pods().createNew()
             .withNewMetadata()
               .withName(kubernetesAppId)
-              .withLabels(selectors)
+              .addToLabels(resolvedSelectors)
```
Comment (ash211) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: ok to make this method change here but not in other places?

**Review by ash211 — COMMENTED** [review]
I'd also like to see a test ensuring that these make it out to the resulting pods (if minikube supports that).

And if you pass an empty value, does it behave as expected?

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala (hunk 2 above)
Comment (mccheah) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: In this context it's fine but this change is accidental and probably should be reverted to reduce the unnecessary diff.

**Comment by mccheah** [comment]
@ash211 good call - tests caught some subtle mistakes in the logic. Fixed all the comments.

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesSuite.scala [hunk 3]
```diff
@@ -155,10 +155,42 @@ private[spark] class KubernetesSuite extends SparkFunSuite with BeforeAndAfter {
...
+    // We can't match all of the selectors directly since one of the selectors is based on the
+    // launch time.
+    assert(driverPodLabels.size == 3, "Unexpected number of pod labels.")
+    assert(driverPodLabels.containsKey("driver-launcher-selector"), "Expected driver launcher" +
+      " selector label to be present.")
+    assert(driverPodLabels.get("label1") == "label1value", "Unexpected value for label1")
+    assert(driverPodLabels.get("label2") == "label2value", "Unexpected value for label2")
+  }
+
+  test("Run driver with custom labels") {
```
Comment (mccheah) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesSuite.scala]: Ah this is a mistake in how the tests were adjusted.

**Review by ash211 — APPROVED** [review]
LGTM -- @foxish thoughts?

**Comment by foxish** [comment]
LGTM.

**Pull request merged**